### DISCUSSION TITLE
Youtube Links

### DISCUSSION THREAD

**Discussion by ghost** [FreeTubeApp/FreeTube#3290] [Q&A]
Hi. 
Why do you use these shortened links (youtu.be) when clicking share and copy link and not the full URL?
I would highly prefer to change it, or at least let the client user choose in the settings. 

Greetings

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
YouTube uses those links for sharing videos. It's not that we chose between one and the other. We just grab the data that is provided.

**Comment by ghost** [comment]
I didn't know, because I've never used youtube natively. 

Can we have full URLs instead?

**Reply by efb4f5ff-1298-471a-8973-3d47447115dc** [reply]
I mean i dont see why we cant but i dont see why though. How will providing the full URL be any better than the shortened one?

Edit: maybe there is a reason why we cant do it but i cant think of any

**Comment by ghost** [comment]
There are a lot of whys: 
https://en.wikipedia.org/wiki/URL_shortening#Shortcomings

**Reply by efb4f5ff-1298-471a-8973-3d47447115dc** [reply]
None of these whys seem to be applicable to YouTube's shortened urls. Lemme rephrase the question here; Why would u want the full urls?

**Comment by ghost** [comment]
What? 
Of course they fit, for example: 
„A short URL obscures the target address and can be used to redirect to an unexpected site.“
Service shutdown, top level domain unknown, because you can not know, you must trust google... and so on.

**Reply by absidue** [reply]
Both of those two aren't applicable, there is no way YouTube would shutdown the youtu.be domain any time soon, if they did they would break almost every single link to YouTube videos that people have shared in the last few years. All of those shortcomings apply to 3rd party link shortening services, not to official link shortening services.

Also it doesn't obscure the target address, youtu.be, clearly says YouTube.

**Comment by ghost** [comment]
Short-URL are good for nothing. 
They are just one example of silly trends on the bullshit web.

**Comment by ghost** [comment]
<del>Discussing with youtube fanboys leads to nothing... </del>
How can anyone really trust that shitty company? 
<del>However please leave us together with the bullshit web!</del>

**Reply by absidue** [reply]
We aren't fanboying/fangirling over YouTube. We just won't change the default behaviour of functionality in FreeTube, to be worse, just because of a one hostile person's personal opionions.

The short YouTube links are easier to type and take up less space in messages, while providing the same functionality. So most people prefer them.

We could add an setting to allow you to copy the long links, if that's something you really want, we'll also happily accept pull requests that add said setting.

**Reply by ghost** [reply]
> We aren't fanboying/fangirling over YouTube. We just won't change the default behaviour of functionality in FreeTube, to be worse, just because of a one hostile person's personal opionions.
> 
> The short YouTube links are easier to type and take up less space in messages, while providing the same functionality. So most people prefer them.
> 
> We could add an setting to allow you to copy the long links, if that's something you really want, we'll also happily accept pull requests that add said setting.

@absidue I am sorry, that was way over the top... I edited my previous post. 
I am just so pissed by all these new things that pop up on the internet these days and that are not improvements, but marked as it... 

But the advantages you mentioned are not really advantages: people normally don't type the urls, they just copy and paste them (with a share button for example). 
And the less space... are you using some kind of messenger or mail hoster that restricts the amount of characters? 

I never share short links, no matter where they are from. 
And when I want to share youtube links, I always have to edit them manually before sending them.

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
Hi @hacksenwerk after some internal discussion we came to the conclusion that there is no benefit to providing the full URL. The short version https://youtu.be points to https://youtube.com and is owned by YouTube. This isnt some kind of third party URL shortner that anyone can create like http://bit.ly
